includes/wikia/api/ArticlesApiController.class.php

```diff
@@ -358,7 +358,7 @@ public function getNew(){
 			$ns = array_unique( $ns );
 		}
 
-		$key = self::getCacheKey( self::NEW_ARTICLES_CACHE_ID, '', [ implode( '-', $ns ) , $minArticleQuality ] );
+		$key = self::getCacheKey( self::NEW_ARTICLES_CACHE_ID, '', [ implode( '-', $ns ) , $minArticleQuality, $limit ] );
```

Comment:
It was actually the original implementation. This is why we needed to ask ImageServing for all of ids. I'll try to pull out the $limit distribution to see if change to this method actually makes any sense.